Fix/#2755 signaling the data source that has a problem

Limraj opened the pull request.

Limraj committed:
#2755 Signaling the Data Source that has a problem -
Created class: DataPointUnreliableUtils;

Signaling added in:
*  Meta Data Source;
*  SNMP;
*  HTTP Retriever;
*  BACnet I/P;
*  DNP3 (IP/Serial);

Signaling use DataPointUnreliableUtils in:
* Modbus IP/Serial;
* Viconics;
* Pachube;
* MQTT/AMQP

Limraj committed:
#2755 Signaling the Data Source that has a problem - corrected signaling on graphical views

Limraj committed:
#2755 Signaling the Data Source that has a problem - corrected signaling on graphical views

Patrykb0802 reviewed (approved):
Tested scenario from issue, results:
- Disabling virtual_ds - Active flag - 
'meta_ds': A context point is unavailable in point "meta_dp appears
- Enbaling, flag goes RTN

Limraj committed:
#2755 Signaling the Data Source that has a problem
- fixed feature refresh point value

Limraj committed:
Merge remote-tracking branch 'origin/release/2.7.7' into fix/#2755_Signaling_the_Data_Source_that_has_a_problem

Limraj committed:
#2755 signaling the data source that has a problem:
- DataPointUnreliableUtilsTest.java;
- use methods returnToNormal and raiseEvent from DataSourceRT;
- added methods: LoggingUtils.info, LoggingUtils.dataPointInfo, LoggingUtils.dataPointRtInfo, PollingDataSource.getDataPoints, EventDataSource.getDataPoints, DataSourceRT.getDataPoints;

Patrykb0802 reviewed (approved):
Tested scenario from issue #2755.
Results:
Data point raises an alarm telling that data point from context is unavailable.
<img width="630" alt="image" src="https://github.com/SCADA-LTS/Scada-LTS/assets/56111976/7c4a9426-83b7-4a90-bff5-c4ddad62b6e9">

Limraj committed:
Merge remote-tracking branch 'origin/release/2.7.7' into fix/#2755_Signaling_the_Data_Source_that_has_a_problem

Limraj committed:
#2755 signaling the data source that has a problem:
- fixed Data Source use serial port - System Exception;

Limraj committed:
#2755 Signaling the data source that has a problem:
- corrected terminate method for: Alpha2DataSource, Dnp3DataSource, DrStorageHt5bDataSource, IEC101DataSource, OPCDataSource, RadiuinoEventDataSource, RadiuinoPollingDataSource;
- fixed reset unreliable HttpImageDataSourceRT when error retriever;
- refactor DataPointUnreliableUtils;
- fixed RuntimeManager.startDataPoint;

Limraj committed:
#2798 Corrected History Annotation Change Point Value By User:
- corrected terminate method for: Alpha2DataSource, Dnp3DataSource, DrStorageHt5bDataSource, IEC101DataSource, OPCDataSource, RadiuinoEventDataSource, RadiuinoPollingDataSource;
- fixed reset unreliable HttpImageDataSourceRT when error retriever;
- refactor DataPointUnreliableUtils;
- fixed RuntimeManager.startDataPoint;

Limraj committed:
#2755 Signaling the data source that has a problem:
- added tests: InitializeDataSourceRtTest, InitializeWithErrorsDataSourceRtTest;
- corrected test: DataPointUnreliableUtilsTest;
- replacing class CommPortIdentifier from RXTXcomm.jar with a new class ScadaCommPortIdentifier: ASCIISerialDataSource, DrStorageHt5bDataSource, RadiuinoEventDataSource, RadiuinoPollingDataSource;
- corrected signaling errors in: BACnetIPDataSourceRT, HttpReceiverDataSourceRT, ModbusDataSource, NmeaDataSourceRT, PachubeDataSourceRT, PersistentDataSourceRT, SnmpDataSourceRT, ViconicsDataSourceRT;
- try catch Throwable;
- change logging level from WARN to ERROR for dataSourceRtLoggingLevel, dataPointRtLoggingLevel (log4j2.xml);
- refactoring DataPointUnreliableUtils;
- added methods: DataPointRT.isUnreliable, DataPointRT.isSetUnreliable, DataSourceUtils.checkInitialized;

Limraj committed:
#2755 Signaling the data source that has a problem:
- added tests: InitializeDataSourceRtTest, InitializeWithErrorsDataSourceRtTest;
- corrected test: DataPointUnreliableUtilsTest;
- replacing class CommPortIdentifier from RXTXcomm.jar with a new class ScadaCommPortIdentifier: ASCIISerialDataSource, DrStorageHt5bDataSource, RadiuinoEventDataSource, RadiuinoPollingDataSource;
- corrected signaling errors in: BACnetIPDataSourceRT, HttpReceiverDataSourceRT, ModbusDataSource, NmeaDataSourceRT, PachubeDataSourceRT, PersistentDataSourceRT, SnmpDataSourceRT, ViconicsDataSourceRT;
- try catch Throwable;
- change logging level from WARN to ERROR for dataSourceRtLoggingLevel, dataPointRtLoggingLevel (log4j2.xml);
- refactoring DataPointUnreliableUtils;
- added methods: DataPointRT.isUnreliable, DataPointRT.isSetUnreliable, DataSourceUtils.checkInitialized;

Limraj committed:
#2755 Signaling the data source that has a problem:
- corrected test: InitializeWithErrorsDataSourceRtTest;
- change type catch from IOException to Throwable;

Limraj committed:
#2755 Signaling the data source that has a problem:
- corrected test: InitializeWithErrorsDataSourceRtTest;
- change type catch from IOException to Throwable;

Limraj committed:
#2755 Signaling the data source that has a problem:
- corrected test: InitializeWithErrorsDataSourceRtTest for VMStatDataSourceRT - update PrepareForTest annotation, if dataSourceRT instanceof VMStatDataSourceRT, for MessagingDataSourceRT with MQTT throw MessagingServiceException after invoke MessageService.open;

Limraj committed:
#2755 Signaling the data source that has a problem:
- corrected test: InitializeWithErrorsDataSourceRtTest for VMStatDataSourceRT - update PrepareForTest annotation, if dataSourceRT instanceof VMStatDataSourceRT, for MessagingDataSourceRT with MQTT throw MessagingServiceException after invoke MessageService.open;

Limraj committed:
#2755 Signaling the data source that has a problem:
- fixed generate many events by ModbusDataSource;
- corrected test InitializeDataSourceRtTest;
- refactor InitializeDataSourceRtMockUtils;

Limraj committed:
#2755 Signaling the data source that has a problem:
- fixed generate many events by ModbusDataSource;
- corrected test InitializeDataSourceRtTest;
- refactor InitializeDataSourceRtMockUtils;

Limraj committed:
#2755 Signaling the data source that has a problem:
- added class: RuntimeMockUtils

Limraj committed:
#2755 Signaling the data source that has a problem:
- added class: RuntimeMockUtils

Limraj committed:
#2755 Signaling the data source that has a problem:
- fixed generate many events by: Pop3DataSourceRT, PachubeDataSourceRT, NmeaDataSourceRT, HttpRetrieverDataSourceRT, BACnetIPDataSourceRT, change to active events;
- change logging fatal to debug level in methods: MBusDataSourceRT.initialize(), MBusDataSourceRT.terminate();
- fixed DataPointUnreliableUtilsTest, InitializeDataSourceRtTest, InitializeWithErrorsDataSourceRtTest, mocking TimeoutTask;

Limraj committed:
#2755 Signaling the data source that has a problem:
- fixed generate many events by: Pop3DataSourceRT, PachubeDataSourceRT, NmeaDataSourceRT, HttpRetrieverDataSourceRT, BACnetIPDataSourceRT, change to active events;
- change logging fatal to debug level in methods: MBusDataSourceRT.initialize(), MBusDataSourceRT.terminate();
- fixed DataPointUnreliableUtilsTest, InitializeDataSourceRtTest, InitializeWithErrorsDataSourceRtTest, mocking TimeoutTask;

Limraj committed:
Merge branch 'fix/#2755_Signaling_the_Data_Source_that_has_a_problem' of https://github.com/SCADA-LTS/Scada-LTS into fix/#2755_Signaling_the_Data_Source_that_has_a_problem

Limraj committed:
#2755 Signaling the data source that has a problem:
- fixed generate many events by SqlDataSourceRT;

Limraj committed:
#2755 Signaling the data source that has a problem:
- fixed export DrStorageHt5bDataSourceVO by removed @JsonRemoteProperty from validate method;

Limraj committed:
Merge branch 'release/2.7.7' into fix/#2755_Signaling_the_Data_Source_that_has_a_problem

Limraj committed:
Merge branch 'release/2.7.7' into fix/#2755_Signaling_the_Data_Source_that_has_a_problem

Limraj committed:
Merge branch 'release/2.7.7.1' into fix/#2755_Signaling_the_Data_Source_that_has_a_problem

Limraj committed:
Merge branch 'release/2.7.7.1' into fix/#2755_Signaling_the_Data_Source_that_has_a_problem
